src/imp/windows.rs

```diff
@@ -0,0 +1,166 @@
+// Distributed under the OSI-approved BSD 3-Clause License.
+// See accompanying LICENSE file for details.
+
+// Implemented by JR Andreassen jr at prtgconsultants com
+// references: 
+//   https://github.com/denisbrodbeck/machineid 
+//   https://github.com/jojonv/rust-win32error/blob/master/src/lib.rs
+
+use winreg::enums::{
+    HKEY_LOCAL_MACHINE, KEY_READ,
+};
+use log::debug;
+use thiserror::Error;
+use uuid::{self, Uuid};
+use std::io::{self};
+
+use rust_win32error::Win32Error;
+
+#[derive(Debug, Error)]
+enum Error {
+    #[error("I/O error")]
+    Io {
+        #[from]
+        source: io::Error,
+    },
+    #[error("W32 error")]
+    W32Err {
+        #[from]
+        source: rust_win32error::Win32Error,
+    },
+    #[error("parse error")]
+    Parse {
+        #[from]
+        source: uuid::Error,
+    },
+}
+
+
+const WIN_REG_BASE_KEY   :&'static str = r"SOFTWARE\\Microsoft\\Cryptography";
+const WIN_REG_BASE_VALUE :&'static str = "MachineGuid";
+
+fn get_machine_id_impl(key_path: &str, key_name: &str) -> Result<Uuid, Error> {
+    let hklm = winreg::RegKey::predef(HKEY_LOCAL_MACHINE);
+    let subkey = //hklm.open_subkey_with_flags(key_path, KEY_READ)?;
+    match hklm.open_subkey_with_flags(key_path, KEY_READ)
+    { // "Failed to open subkey"
+        Ok(val) => val,
+        Err(err) => {
+            //let mut w32err = Win32Error::init_from_error_code(err.);
+            debug!("get_machine_id_impl: Failed to read machine UUID W32 {:?}", err);
+            return Err(err.into());  
+//            return Err(w32err.into());  
+        },
+    };
+
+    let machine_uuid: String = 
+      match subkey.get_value(key_name)
+      {
+          Ok(value) => value,
+          Err(err) => {
+            let w32err = Win32Error::new();
+            if w32err.get_error_code() != 0 
+            {   
+                debug!("get_machine_id_impl: Failed to read machine UUID W32 {:?}", w32err);
+                return Err(w32err.into());
+            }
+            debug!("get_machine_id_impl: Failed to read machine UUID W32 {:?}", err);
+            return Err(err.into());
+          },
+      };
+    Ok(Uuid::parse_str(&machine_uuid)?)
+}
+
+pub fn get_machine_id() -> Option<Uuid> {
+    match get_machine_id_impl(WIN_REG_BASE_KEY, WIN_REG_BASE_VALUE) {
+        Ok(uuid) => Some(uuid),
+        Err(err) => {
+            debug!("{:?}", err);
+            None
+        },
+    }
+}
+
+// ############################################################################################
```

Comment:
Please remove these "barrier" comments. I find them to just be noise because keeping them up-to-date is done by approximately no one.

Reply:
Unfortunate, I think they make the code easier to read...
But, your project, your rules...